### ISSUE TITLE
Operation with && in event handlers via props causes the 'outside a createRoot' error

### ISSUE THREAD

**Issue by timothyallan** [solidjs/solid#1715]
### Describe the bug

It was suggested that the Discord support thread I raised should be logged as an issue, so here goes:

Why does button 1 generate a warning? Doesn't it just evaluate to boolean anyways?

### Your Example Website or App

 https://playground.solidjs.com/anonymous/d5fe0e41-701e-4360-b991-867f9e57811f

### Steps to Reproduce the Bug or Issue

Click button 1, check out the console warning.

### Expected behavior

No warning.

### Platform

MacOS, all browsers

**Comment by atk** [comment]
Events, timeouts, promises are all out of the normal execution context and therefore, also out of the reactive root, which allows registering signals as dependencies.

A simple solution is to use an event signal to move the event into the reactive system:

```ts
const [ev, setEv] = createSignal();
createEffect(on(ev, () => {
  console.log(otherSignal());
}, { defer: true }));
return <div onClick={setEv}>...</div>;
```

**Comment by timothyallan** [comment]
Thanks for the suggestion. Using the `Boolean` wrapper (button 2 in the playground) seems to be a substantially less involved solution, I'll stick with that method I guess.

it still seems odd from a DX perspective that using one reactive signal 'works', but &&'ing 2 causes the issue. 

Sending in a combined reactive function works as well, so you can combine them beforehand: https://playground.solidjs.com/anonymous/84b38f1c-79a4-4624-8164-00d83062a36c

**Comment by ryansolid** [comment]
Yeah thinking about this. This is a bug. I changed some stuff in the transform a few versions back to fix one problem but introduced this one. We will look at getting this fixed up.

**Comment by ryansolid** [comment]
Hmm.. I can't think of a way with how our current reactivity works to solve both issues. Either we process the prop too eagerly which isn't acceptable (and can break things like hydration) or we have this issues. We could just drop this optimization which has performance implications. We basically need a lazy memo that self releases to fix this. Putting this firmly in Solid 2.0 territory.

**Comment by ryansolid** [comment]
fixed in 2.0
